packages/core/admin/admin/src/pages/SettingsPage/pages/AuditLogs/ListView/hooks/useAuditLogsData.js

```diff
@@ -0,0 +1,49 @@
+import { useQueries } from 'react-query';
+import { useNotification, useFetchClient } from '@strapi/helper-plugin';
+import { useLocation } from 'react-router-dom';
+
+const useAuditLogsData = ({ canRead }) => {
+  const { get } = useFetchClient();
+  const { search } = useLocation();
+  const toggleNotification = useNotification();
+
+  const fetchAuditLogsPage = async ({ queryKey }) => {
+    const search = queryKey[1];
+    const { data } = await get(`/admin/audit-logs${search}`);
+
+    return data;
+  };
+
+  const fetchAllUsers = async () => {
+    const { data } = await get(`/admin/users`);
+
+    return data;
+  };
+
+  const queryOptions = {
+    enabled: canRead,
+    keepPreviousData: true,
+    retry: false,
+    staleTime: 1000 * 20,
+    onError() {
+      toggleNotification({
+        type: 'warning',
+        message: { id: 'notification.error', defaultMessage: 'An error occured' },
+      });
+    },
+  };
+
+  const [auditLogsData, userData] = useQueries([
+    { queryKey: ['auditLogs', search], queryFn: fetchAuditLogsPage, ...queryOptions },
+    { queryKey: ['auditLogsUsers'], queryFn: fetchAllUsers, ...queryOptions },
+  ]);
+
+  const { data: users, isLoading: isLoadingUsers } = userData;
+  const { data: auditLogs, isLoadingAuditLogs } = auditLogsData;
+
+  const isLoading = isLoadingAuditLogs || isLoadingUsers;
```

Comment:
I'm wondering if it's necessary to block the rendering of the whole page for the users query . Do you think loading them in the background would be viable? 🤔

Reply:
> This is okay but yes the select thing would be cleaner. Maybe it's because of a version issue? The docs are for the latest react-intl v6, we have v5

I'll take a deeper look but I did check and if I remember correctly the select has been available since early v5